misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java

```diff
@@ -182,7 +182,16 @@ public List<String> skipList() {
                 "client_resource_access_action_source_05.bal", "client_resource_access_action_source_06.bal",
                 "resiliency_source_05.bal", "regexp_constructor_source_26.bal", "regexp_constructor_source_28.bal",
                 "regexp_constructor_source_44.bal", "regexp_constructor_source_47.bal",
-                "regexp_constructor_source_52.bal", "regexp_constructor_source_53.bal", "completion_source_07.bal");
+                "regexp_constructor_source_52.bal", "regexp_constructor_source_53.bal", "completion_source_07.bal",
+                "do_stmt_source_14.bal", "do_stmt_source_15.bal", "do_stmt_source_16.bal", "do_stmt_source_17.bal",
+                "forEach_stmt_source_31.bal", "foreach_stmt_source_32.bal", "forEach_stmt_source_33.bal",
+                "foreach_stmt_source_34.bal", "lock_stmt_source_15.bal", "lock_stmt_source_16.bal",
+                "lock_stmt_source_17.bal", "lock_stmt_source_18.bal", "match_stmt_source_28.bal",
+                "match_stmt_source_29.bal", "match_stmt_source_30.bal", "match_stmt_source_31.bal",
+                "retry_stmt_source_12.bal", "retry_stmt_source_13.bal", "retry_stmt_source_14.bal",
+                "retry_stmt_source_15.bal", "transaction_stmt_source_09.bal", "transaction_stmt_source_10.bal",
+                "transaction_stmt_source_11.bal", "transaction_stmt_source_12.bal", "while_stmt_source_11.bal",
+                "while_stmt_source_12.bal", "while_stmt_source_13.bal", "while_stmt_source_14.bal");
```

Comment:
Shall we not skip the recovery tests via the formatter? Formatter is supposed to format on top of negative syntax as well. We can skip only the ones having a lot of recoveries involved/ uncommon scenarios. e.g. parser resiliency tests.

Reply:
For example, `while_stmt_source_13.bal` has the following diff after formatting. We can keep the test in the formatted form. 

![Screenshot from 2023-04-24 12-10-47](https://user-images.githubusercontent.com/39232462/233918677-5d91f988-dd21-4d35-8269-2a65b7c6cb44.png)

Reply:
if it helps, you can apply the following patch https://gist.github.com/lochana-chathura/e8736c375eadaa62a42387a289373222 to see the diff. 
1.Apply the patch. 
2. Copy the code sample to both `module_variable_declaration_1.bal` named files.
3. run `test()` in the commented line. 
4. If there's a diff you will get something like below. Click on `Click to see the difference` to get the diff. 

![Screenshot from 2023-04-24 12-21-10](https://user-images.githubusercontent.com/39232462/233920468-749aeab6-4e8f-45fa-b44d-80179bee0e46.png)